torch/_dynamo/guards.py

```diff
@@ -725,8 +726,40 @@ def compile_check_fn(
                 local_builder.tensor_check_examples
                 + global_builder.tensor_check_examples
             )
+            dynamic_dims_sizes = None
+            dynamic_dims_strides = None
+            if config.dynamic_shapes:
+
+                def convert(size_or_stride):
+                    converted = []
+                    for dim in size_or_stride:
+                        if is_concrete_int(dim):
+                            converted.append(int(dim))
+                        else:
+                            converted.append(None)
+                    return converted
+
+                dynamic_dims_sizes = [
+                    convert(
+                        self.output_graph.tracing_context.fake_mode.from_tensor(
```

Comment:
We can do this in a follow up, but we should add an api to fake mode which forces a cache hit (and errors if it's not in cache) and use it here

Reply:
Agreed